diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/CollapseIfStatementsRule.kt

```diff
@@ -0,0 +1,159 @@
+package org.cqfn.diktat.ruleset.rules.chapter3
+
+import com.pinterest.ktlint.core.ast.ElementType.BINARY_EXPRESSION
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.cqfn.diktat.ruleset.rules.DiktatRule
+import org.cqfn.diktat.ruleset.utils.KotlinParser
+
+import com.pinterest.ktlint.core.ast.ElementType.BLOCK_COMMENT
+import com.pinterest.ktlint.core.ast.ElementType.EOL_COMMENT
+import com.pinterest.ktlint.core.ast.ElementType.IF
+import com.pinterest.ktlint.core.ast.ElementType.KDOC
+import com.pinterest.ktlint.core.ast.ElementType.LBRACE
+import com.pinterest.ktlint.core.ast.ElementType.OPERATION_REFERENCE
+import com.pinterest.ktlint.core.ast.ElementType.RBRACE
+import com.pinterest.ktlint.core.ast.ElementType.THEN
+import com.pinterest.ktlint.core.ast.ElementType.WHITE_SPACE
+import org.cqfn.diktat.common.config.rules.RuleConfiguration
+import org.cqfn.diktat.common.config.rules.getRuleConfig
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.KtBlockExpression
+import org.jetbrains.kotlin.psi.KtIfExpression
+
+/**
+ * Rule for redundant nested if-statements, which could be collapsed into a single one
+ */
+class CollapseIfStatementsRule(configRules: List<RulesConfig>) : DiktatRule(
+    "collapse-if",
+    configRules,
+    listOf(
+        Warnings.COLLAPSE_IF_STATEMENTS
+    )
+) {
+    private val configuration by lazy {
+        CollapseIfStatementsConfiguration(
+            configRules.getRuleConfig(Warnings.COLLAPSE_IF_STATEMENTS)?.configuration ?: emptyMap()
+        )
+    }
+    override fun logic(node: ASTNode) {
+        if (node.elementType == IF) {
+            process(node)
+        }
+    }
+
+    private fun process(node: ASTNode)  {
+        val startCollapseFromLevel = configuration.startCollapseFromNestedLevel
+
+        var currNode = findNestedIf(node) ?: return
+        var nestedLevel = 2
+        var nestedIfNode :ASTNode?
+        while (true) {
+            nestedIfNode = findNestedIf(currNode)
+            if (nestedIfNode != null) {
+                nestedLevel++
+                currNode = nestedIfNode
+            } else {
+                break
+            }
+        }
+        if (nestedLevel < startCollapseFromLevel) {
+            return
+        }
+        // Since the external `if` statement is not the direct parent,
+        // we need multiple steps to take the required one
+        // BLOCK -> THEN -> IF
+        var currParentNode = currNode.treeParent.treeParent.treeParent
+        // Now collapse in reverse order
+        while (nestedLevel != 1 && isFixMode) {
```

Comment:
```suggestion
        while (nestedLevel != 1) {
```
No need to check `isFixMode` manually, it's handled inside `warnAndFix`